Percy/AppPercy.cs

```
@@ -24,9 +24,16 @@ public AppPercy(Object driver)
 
     internal void setValues(IPercyAppiumDriver percyAppiumDriver)
     {
-      this.percyOptions = new PercyOptions(percyAppiumDriver);
-      this.isPercyEnabled = CliWrapper.Healthcheck();
-      this.sessionId = percyAppiumDriver.sessionId();
+      try
+      {
+        this.percyOptions = new PercyOptions(percyAppiumDriver);
+        this.isPercyEnabled = CliWrapper.Healthcheck();
+        this.sessionId = percyAppiumDriver.sessionId();
+      } 
+      catch (Exception e)
+      {
+        throw new Exception($"Error fetching sessionId: {e.ToString()}");

```

Dont add specific exception here, if its referring to sessionId just at it inside `percyAppiumDriver.sessionId();` so that you dont need to throw this exception at all places where you are using `sessionId` function

---

Actually redisUtils will throw, since its a hard stop we don’t need to do anything. i moved it here initially when I was doing percyOptions thing. removing from here.